File: Practice/AllDomains/Tutorials/30DaysOfCode/Day9Recursion.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution 
{
	/*	author:@shivkrthakur  */
    static void Main(String[] args) 
	{
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int x = Convert.ToInt32(Console.ReadLine().Trim());
        Console.WriteLine(Factorial(x));
    }
    
    public static int Factorial(int x)
    {
        if(x == 1) return x;
        return x * Factorial(x-1);
    }
}

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day9Recursion.js
/*	author:@shivkrthakur  */
function processData(input) {
    //Enter your code here
    var n = parseInt(input);
    console.log(factorial(n));
}

function factorial(input){
   if(input == 1) return 1;
   return input * factorial(input - 1);
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
